dev-example/src/example.test.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  createMockMachineRunner,
+  createMockState,
+  createMockStateOpaque,
+} from '@actyx/machine-runner/lib/test-utils'
+import { Bid, BidData, BidderID, Cancelled, Passenger, Requested } from './machines/index.js'
+import { toPrettyJSONString } from './UIMachineCommon.js'
+import { isTaxiRideCancelEnabled } from './UIMachinePassenger.js'
+
+describe('State transformation tests', () => {
+  it('should verify if state transformation as written by the consumer code is correct', () => {
+    const requestDestination = 'destination'
+    const requestPickup = 'pickup'
+    const bidTime = new Date()
+    const bidPrice = 1
+    const bidderId = 'bidderId'
+
+    const machineRunner = createMockMachineRunner(Passenger.Initial, void 0)
+
+    machineRunner.test.feed([
+      Requested.make({
+        destination: requestDestination,
+        pickup: requestPickup,
+      }),
+      Bid.make({
+        price: bidPrice,
+        time: bidTime.toISOString(),
+      }),
+      BidderID.make({
+        id: bidderId,
+      }),
+    ])
+
+    const auction1 = machineRunner.test.assertAs(Passenger.Auction, (auction) => {
+      expect(auction.payload.bids.at(0)).toEqual({
+        bidderID: bidderId,
+        time: bidTime,
+        price: bidPrice,
+      } as BidData)
+
+      return auction
+    })
+
+    // Alternative usage example of `assertAs`
+    const auction2 = machineRunner.test.assertAs(Passenger.Auction)
+    expect(auction2.payload.bids.at(0)).toEqual({
+      bidderID: bidderId,
+      time: bidTime,
+      price: bidPrice,
+    } as BidData)
+
+    // Comparing arrays of objects that has a Date member is broken in jest. This doesn't work.
```

Comment:
If there is a GitHub issue on Jest's repo, a link would be useful to keep track if it gets fixed